ThunderCloud/CollectionItemViewCell.swift

```diff
@@ -10,144 +10,210 @@ import UIKit
+        return [titleLabel, subtitleLabel]
+    }
+    
     override open func awakeFromNib() {
-        
         super.awakeFromNib()
-        titleLabel.textColor = ThemeManager.shared.theme.cellTitleColor
         
-        imageBackgroundView.clipsToBounds = false
         clipsToBounds = false
         contentView.clipsToBounds = false
-        
-        imageContainerView.cornerRadius = 38
-        imageBackgroundView.cornerRadius = 38
-        imageBackgroundView.clipsToBounds = false
-        
-        imageBackgroundView.shadowRadius = 36
-        imageBackgroundView.shadowColor = .black
-        imageBackgroundView.shadowOffset = CGPoint(x: 0, y: 10)
-        imageBackgroundView.shadowOpacity = 0.1
-        
-        imageBackgroundView.borderWidth = 1.0/UIScreen.main.scale
-        imageBackgroundView.borderColor = UIColor(white: 0.0, alpha: 0.04)
     }
     
     /// Calculates the size of the collection list item for the given item
     ///
     /// - Parameter item: The item which will be rendered
     /// - Returns: The size the items content will occupy
     public class func size(for item: CollectionCellDisplayable) -> CGSize {
+        var width = Constants.imageBackgroundViewSize
+        var height = Constants.imageBackgroundViewSize
         
-        let enabled = item.enabled
-        
-        let cellWidthPadding = CollectionItemViewCell.cellPadding.left + CollectionItemViewCell.cellPadding.right
-        let cellHeightPadding = CollectionItemViewCell.cellPadding.top + CollectionItemViewCell.cellPadding.bottom
-        
-        guard let title = item.itemTitle, !title.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
-            return CGSize(width: CollectionItemViewCell.imageSize.width + cellWidthPadding, height: CollectionItemViewCell.imageSize.height + cellHeightPadding)
-        }
-        
-        let widthLabel = CollectionItemViewCell.widthCalculationLabel
-        widthLabel.text = title
-        widthLabel.font = ThemeManager.shared.theme.dynamicFont(ofSize: 13, textStyle: .footnote, weight: enabled ? .bold : .regular)
-        widthLabel.numberOfLines = 1
-        widthLabel.sizeToFit()
-        
-        let labelPadding = CollectionItemViewCell.labelPadding.left + CollectionItemViewCell.labelPadding.right
-        let contentWidth = max(CollectionItemViewCell.imageSize.width, widthLabel.frame.width + labelPadding)
-        let width = contentWidth + cellWidthPadding
-        
-        let labelHeightPadding = CollectionItemViewCell.labelPadding.bottom + CollectionItemViewCell.labelPadding.top
-        let height = cellHeightPadding + CollectionItemViewCell.imageSize.height + labelHeightPadding + CollectionItemViewCell.labelImageSpacing + widthLabel.frame.height
+        includeLabelDimensions(text: item.title, style: item.titleStyle,
+                               width: &width, height: &height)
+        includeLabelDimensions(text: item.expiryDateString, style: item.expiryStyle,
```

Comment:
Don't see `inout` used much myself, but this is really nice way to make this more functional!

Reply:
Yeah really rare tbh!